ToolBox/Program.cs

```diff
@@ -0,0 +1,190 @@
+                                                                                     GetParameterOrExit(optionsObject, "resolution"));
+                        break;
+                    case "kdc":
+                    case "kaikodataconverter":
+                        KaikoDataConverterProgram.KaikoDataConverter(GetParameterOrExit(optionsObject, "market"),
+                                                                     GetParameterOrExit(optionsObject, "tick-type"),
+                                                                     GetParameterOrExit(optionsObject, "source-dir"));
+                        break;
+                    case "nmdc":
+                    case "nsemarketdataconverter":
+                        NseMarketDataConverterProgram.NseMarketDataConverter(GetParameterOrExit(optionsObject, "source-dir"),
+                                                                             GetParameterOrExit(optionsObject, "destination-dir"));
+                        break;
+                    case "qqc":
+                    case "quantquoteconverter":
+                        QuantQuoteConverterProgram.QuantQuoteConverter(GetParameterOrExit(optionsObject, "destination-dir"),
+                                                                       GetParameterOrExit(optionsObject, "source-dir"),
+                                                                       GetParameterOrExit(optionsObject, "resolution"));
+                        break;
+                    case "cug":
+                    case "coarseuniversegenerator":
+                        CoarseUniverseGeneratorProgram.CoarseUniverseGenerator();
+                        break;
+                    default:
+                        PrintMessageAndExit("ERROR: Unrecognized --app value");
+                        break;
+                }
+            }
+        }
+        private static IList<string> DictionaryObjectToList(object dictionaryObject)
+        {
+            IList<string> list = new List<string>();
+            var dictionary = dictionaryObject as Dictionary<string, object>;
+            foreach (var item in dictionary)
+            {
+                list.Add(item.Key);
+            }
+            return list;
+        }
+
+        private static void PrintMessageAndExit(string message = "")
+        {
+            if (!message.IsNullOrEmpty())
+            {
+                Console.WriteLine("\n" + message);
+                Console.WriteLine("Use the '--h' parameter for more information");
+            }
+            Console.WriteLine("\nPress any key to quit");
+            Console.ReadLine();
+            Environment.Exit(0);
+        }
+
+        private static string GetParameterOrExit(IReadOnlyDictionary<string, object> optionsObject, string parameter)
+        {
+            if (!optionsObject.ContainsKey(parameter))
+            {
+                PrintMessageAndExit("ERROR: REQUIRED parameter --" + parameter + "= is missing");
```

Comment:
This one is _definitely_ an error case and should end up with a non-zero exit code.